docs/subnets/case-study-wagmi-upgrade.md

```diff
@@ -0,0 +1,154 @@
+
+## Using Fee Manager
+
+The owner `0x6f0f6DA1852857d7789f68a28bba866671f3880D` can now configure the fees on the Subnet as they see fit. To do that, all that's needed is access to the network, the private key for the newly set manager address and making calls on the precompiled contract.
+
+We will use [Remix](https://remix.ethereum.org) online Solidity IDE and the [Core Browser Extension](https://support.avax.network/en/articles/6066879-core-extension-how-do-i-add-the-core-extension). Core comes with WAGMI network built-in. Metamask will do as well but you will need to [add WAGMI](./wagmi.md#adding-wagmi-to-metamask) yourself.
+
+First we connect Core to WAGMI. Switch on the `Testnet Mode` in `Advanced` page in the hamburger menu:
+
+![Core testnet mode](/img/network-upgrade/core-testnet-mode.png)
+
+And then open the `Manage Networks` menu in the networks dropdown. Select WAGMI there by clicking the star icon:
+
+![Core network selection](/img/network-upgrade/core-network-select.png)
+
+We then switch to WAGMI in the networks dropdown. We are ready to move on to Remix now, so we open it in the browser. First, we check that Remix sees the extension and correctly talks to it. We select `Deploy & run transactions` icon on the left edge, and on the Environment dropdown, select `Injected Provider`. We need to approve the Remix network access in the Core browser extension. When that is done, `Custom (11111) network` is shown:
+
+![Injected provider](/img/network-upgrade/remix-injected-provider.png)
+
+Good, we're talking to WAGMI Subnet. Next we need to load the contracts into Remix. Using 'load from Github' option from the Remix homescreen we load two contracts:
+
+- [IAllowList.sol](https://github.com/ava-labs/subnet-evm/blob/master/contract-examples/contracts/IAllowList.sol)
+- and [IFeeManager.sol](https://github.com/ava-labs/subnet-evm/blob/master/contract-examples/contracts/IFeeManager.sol).
+
+IFeeManager is our precompile, but it references the IAllowList, so we need that one as well. We compile IFeeManager.sol and deploy at the precompile address `0x0200000000000000000000000000000000000003` used on the [Subnet](https://github.com/ava-labs/subnet-evm/blob/master/precompile/params.go#L33).
+
+![Deployed contract](/img/network-upgrade/deployed-contract.png)
+
+Now we can interact with the FeeManager precompile from within Remix via Core. For example, we can use the `getFeeConfig` method to check the current fee configuration. Once we have the new desired configuration for the fees on the Subnet, we can use the `setFeeConfig` to change the parameters:
+
+![setFeeConfig](/img/network-upgrade/setFeeConfig.png)
+
+When we call that method by pressing the `transact` button, a new transaction is posted to the Subnet, and we can see it on [the explorer](https://subnets-test.avax.network/wagmi/block/0xad95ccf04f6a8e018ece7912939860553363cc23151a0a31ea429ba6e60ad5a3):
```

Comment:
Can we add a very clear note that states only the admin address will be able to do this?

Reply:
Could we also put `getFeeConfig` first so that the first example is one users can do without being an admin address?

I don't love doing it twice, but could we do check the current config, set the config (with a warning only admins can do this), and then read the updated config?

Reply:
Great suggestion. I have modified the text based on your review. However, as we did not have two different screenshots of getFeeConfig, I just used words to describe the first occurrence of getFeeConfig.

Reply:
Yes, I wanted to have the getFeeConfig first as well, but we don't have the picture with the previous gas configuration (and I don't know how to get it retroactively), so the before and after gas configs would look the same which would kinda defeat the purpose of the whole thing. :-)